### ISSUE TITLE
"Invalid header name detected" (From XXX) on upgrades

### ISSUE THREAD

**Issue by vladsf** [eventum/eventum#427]
Hi,

I have "Invalid header name detected" issue on 'php bin/upgrade' running and in browser viewing older issues e.g. http://eventumtest01.test.com/view_email.php?ema_id=20&id=236398

I saw #155 issue, but the fix does not solve the trouble on upgrades or views. 

Eventum v3.1.10 - php bin/upgrade:
```
[2018-12-24 15:44:05] db.INFO: skipped maq_id=867359, exception: Line "From xxx.xxx@company.com  Mon Jan  5 13: 05:51 2015"does not match header f
ormat! [] {"file":"/intranet/eventum-3.1.10/upgrade/patches/56_maq_message_id.php","line":52,"class":"Eventum\\Db\\Migrate","function":"{closure}","memory_
usage":"68.5 MB","memory_peak_usage":"74 MB","version":"3.1.10","usr_id":""}
```
The same on upgrade run from 3.2.0 to 3.5.4:
```
[2018-12-25 10:14:15] db.INFO: skipped maq_id=867198, exception: Invalid header name detected [] {"file":"/intranet/eventum-3.5.4/db/migrations/20180306202659_eventum_maq_message_id.php","line":49,"class":"EventumMaqMessageId","function":"up","memory_usage":"42.75 MB","memory_peak_usage":"45.25 MB","version":"3.5.4","usr_id":""}
```
Then it fails in browser (shows empty page) and eventum.log gets: 
```
[2018-12-25 10:34:25] app.ERROR: Uncaught Exception Zend\Mail\Header\Exception\InvalidArgumentException: "Invalid header name detected From name.surname@test.com  Wed Dec 19 08" at /intranet/eventum/vendor/zendframework/zend-mail/src/Header/GenericHeader.php line 57 {"exception":"[object] (Zend\\Mail\\Header\\Exception\\InvalidArgumentException(code: 0): Invalid header name detected From name.surname@test.com  Wed Dec 19 08 at /intranet/eventum/vendor/zendframework/zend-mail/src/Header/GenericHeader.php:57)"} {"url":"/view_email.php?ema_id=20&id=236695","ip":"10.xx.xx.xx","http_method":"GET","server":"eventumtest01.test.com","referrer":"http://eventumtest01.test.com/view.php?id=62010","unique_id":"XCIHsd2LuhHFE9IJ6@2unQAAAAE","file":null,"line":null,"class":null,"function":null,"memory_usage":"11.75 MB","memory_peak_usage":"11.75 MB","version":"3.5.4","usr_id":"107"}
```
The emails in the database are from Eventum 2.3.6, fetched from IMAP server. I am testing upgrade from 2.3.6 to 3.5.4.

One option is to ignore src/Header/GenericHeader.php exceptions, but the impact of this hack is not clear. Also it is not obvious if I shall run bin/upgrade again.

**Comment by vladsf** [comment]
installed Xdebug and got this:
Eventum\Mail\Helper\MailLoader::fixBrokenHeaders is called at /intranet/eventum/src/Mail/Helper/MailLoader.php:114Eventum\Mail\Helper\MailLoader::fixBrokenHeaders is called at /intranet/eventum/src/Mail/Helper/MailLoader.php:121

**Comment by vladsf** [comment]
I suggest this fix to be applied to src/Mail/Helper/MailLoader.php:
[iss427-diff.txt](https://github.com/eventum/eventum/files/2709088/iss427-diff.txt)

**Comment by glensc** [comment]
the patch that fails is from #352 and it's errors can be ignored (as the log says)

**Comment by vladsf** [comment]
Hi @glensc,
Thank yo for the quick reply. I will do the pull request once I am back to keyboard from holidays.

**Comment by glensc** [comment]
update from `2.3.6` to `3.5.4` is huge. good luck with that! :)

btw, `3.5.5` was released few days ago with #423, which may help you.

**Comment by vladsf** [comment]
Hi @glensc, 

I have created a pull request. Eventum tests/data already has a test email for the case. I hope I made the request correctly. Thanks!

**Comment by glensc** [comment]
fixed via #439
